packages/toolpad-core/src/shared/Link.tsx

```diff
@@ -5,15 +5,17 @@ import { RouterContext } from './context';
  * @ignore - internal component.
  */
 
-export interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
+export interface LinkProps extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> {
```

Comment:
I believe this should be a small interface. With only the properties we need for usage inside of Toolpad core.